eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/rpc/BitcoinCoreClient.scala

```diff
@@ -170,19 +170,21 @@ class BitcoinCoreClient(val rpcClient: BitcoinJsonRPCClient) extends OnChainWall
 
   /**
    * Iterate over blocks to find the transaction that has spent a given output.
-   * NB: only call this method when you're sure the output has been spent, otherwise this will iterate over the whole
-   * blockchain history.
+   * NB: this will iterate over the past month of blockchain history, which is resource-intensive.
    *
    * @param blockhash_opt hash of a block *after* the output has been spent. If not provided, we will use the blockchain tip.
    * @param txid          id of the transaction output that has been spent.
    * @param outputIndex   index of the transaction output that has been spent.
    * @return the transaction spending the given output.
    */
   def lookForSpendingTx(blockhash_opt: Option[ByteVector32], txid: ByteVector32, outputIndex: Int)(implicit ec: ExecutionContext): Future[Transaction] = {
-    lookForSpendingTx(blockhash_opt.map(KotlinUtils.scala2kmp), KotlinUtils.scala2kmp(txid), outputIndex)
+    // It isn't useful to look at the whole blockchain history: if the transaction was confirmed long ago, an attacker
+    // will have already claimed all possible outputs and there's nothing we can do about it.
+    val limit = 4 * 720
```

Comment:
Why not make it a parameter of `lookForSpendingTx` ? It would then be easy to add a unit test.

Reply:
Yes, if we agree this feature is desired, I'll add a unit test that calls the override that already has it as an explicit parameter.

Reply:
I don't know what should be the magic value by the way. Should we make it configurable in `eclair.conf`? Should we just use the `max-to-local-delay-blocks`? Or even something smaller than that, that rather represent the maximum amount of time we expect the node to be disconnected from the blockchain?

Reply:
The value is not related to our local delay but to the time since eclair was stopped. So unfortunately I believe we must make it configurable.

Reply:
It is bounded by our local delay though: if the transaction was confirmed more than `to_self_delay` blocks ago, we can't even publish a penalty transaction, our peer has successfully cheated already, so it probably doesn't make sense to even look at it?

But in a way, `to_self_delay` is linked to the maximum time we expect our node to be down, it must always be greater than our downtime to guarantee funds safety.

Reply:
> our peer has successfully cheated already, so it probably doesn't make sense to even look at it?

It's too late for punishment, but we still need to retrieve our delayed local balance (in the anchor outputs case), right?

Reply:
Right, in the case where our peer was honest and simply broadcast the latest commitment, it's still worth finding that transaction to get our funds back!

Reply:
Even if they publish a revoked commitment, we may have a non-zero balance?

Reply:
Right, that's true as well.

Reply:
Done in https://github.com/ACINQ/eclair/pull/2731/commits/1276f2c816ce1fb7bb6d5e91b5bb90ba43dd1668